File: packages/engine/src/runtime/host.ts
/**
 * ③ 宿主信息解析（纯函数，可测）：把宿主提供的「编译期平台字符串」收敛为 HostInfo。
 * 输入 = Tauri CLI 注入的 `TAURI_ENV_PLATFORM`（windows/macos/linux/android/ios，
 * 大小写不敏感）；无法识别/缺省 → `unknown·desktop`（显式未知，不猜）。
 */
import { hostFormOf, type HostInfo, type HostOs } from "../contracts";

/** Tauri 编译期平台 → HostOs（键全部小写，匹配前先归一化） */
const PLATFORM_BY_RAW: Record<string, Exclude<HostOs, "unknown">> = {
  windows: "windows",
  macos: "macos",
  linux: "linux",
  android: "android",
  ios: "ios",
};

export function resolveHost(rawPlatform: string | undefined): HostInfo {
  const key = rawPlatform?.trim().toLowerCase();
  const os: HostOs = (key && PLATFORM_BY_RAW[key]) || "unknown";
  return { os, form: hostFormOf(os) };
}

File: apps/playground/src-tauri/src/host.rs
//! ③ 平台区分（宿主事实源）：把编译目标平台暴露给渲染端。
//!
//! 为什么是 Rust 命令而不是编译期环境变量（`TAURI_ENV_PLATFORM`）：实测 Tauri CLI 的
//! `beforeBuildCommand` 子进程**没有**注入该变量（Android 构建下前端拿到 undefined →
//! 宿主误判为 desktop，iOS 构建同理）→ 编译期路线不可靠；`std::env::consts::OS` 在
//! 对应 target 下恒正确（android/ios/windows/macos/linux），且零依赖、零权限面。
//!
//! 取值与契约 `HostOs` 对齐（见 packages/engine/src/contracts/host.ts）。

/// 当前编译目标平台（`std::env::consts::OS`：windows/macos/linux/android/ios）
#[tauri::command]
pub fn host_platform() -> &'static str {
    std::env::consts::OS
}
